Gems/AtomTressFX/Code/Builders/HairAssetBuilder.cpp

```diff
@@ -0,0 +1,168 @@
+                builderDesc.m_busId = azrtti_typeid<HairAssetBuilder>();
+                builderDesc.m_version = 3;
+                builderDesc.m_createJobFunction =
+                    AZStd::bind(&HairAssetBuilder::CreateJobs, this, AZStd::placeholders::_1, AZStd::placeholders::_2);
+                builderDesc.m_processJobFunction =
+                    AZStd::bind(&HairAssetBuilder::ProcessJob, this, AZStd::placeholders::_1, AZStd::placeholders::_2);
+
+                BusConnect(builderDesc.m_busId);
+
+                AssetBuilderSDK::AssetBuilderBus::Broadcast(
+                    &AssetBuilderSDK::AssetBuilderBusTraits::RegisterBuilderInformation, builderDesc);
+            }
+
+            void HairAssetBuilder::ShutDown()
+            {
+                m_isShuttingDown = true;
+            }
+
+            void HairAssetBuilder::CreateJobs(
+                const AssetBuilderSDK::CreateJobsRequest& request, AssetBuilderSDK::CreateJobsResponse& response)
+            {
+                if (m_isShuttingDown)
+                {
+                    response.m_result = AssetBuilderSDK::CreateJobsResultCode::ShuttingDown;
+                    return;
+                }
+
+                for (const AssetBuilderSDK::PlatformInfo& info : request.m_enabledPlatforms)
+                {
+                    AssetBuilderSDK::JobDescriptor descriptor;
+                    descriptor.m_jobKey = AMD::TFXFileExtension;
+                    descriptor.m_critical = false;
+                    descriptor.SetPlatformIdentifier(info.m_identifier.c_str());
+                    response.m_createJobOutputs.push_back(descriptor);
+                }
+
+                // Set the tfx bone and tfx mesh file as source dependency. This way when tfxbone or tfxmesh file reloaded it will
+                // also trigger the rebuild of the hair asset.
+                AssetBuilderSDK::SourceFileDependency sourceFileDependency;
+                sourceFileDependency.m_sourceFileDependencyPath = request.m_sourceFile;
+                AZ::StringFunc::Path::ReplaceExtension(sourceFileDependency.m_sourceFileDependencyPath, AMD::TFXBoneFileExtension);
+                response.m_sourceFileDependencyList.push_back(sourceFileDependency);
+
+                sourceFileDependency.m_sourceFileDependencyPath = request.m_sourceFile;
+                AZ::StringFunc::Path::ReplaceExtension(sourceFileDependency.m_sourceFileDependencyPath, AMD::TFXMeshFileExtension);
+                response.m_sourceFileDependencyList.push_back(sourceFileDependency);
+
+                response.m_result = AssetBuilderSDK::CreateJobsResultCode::Success;
+            }
+
+            void HairAssetBuilder::ProcessJob(
+                const AssetBuilderSDK::ProcessJobRequest& request, AssetBuilderSDK::ProcessJobResponse& response)
+            {
+                AZ_TracePrintf(AssetBuilderSDK::InfoWindow, "HairAssetBuilder Starting Job.\n");
```

Comment:
These traceprintfs are good, but it begs the question, at some point in your future you will probably be sitting down looking at a log file of a failed asset build and the only clues you'll have is your traceprintfs in the log.  Is this sufficient information that you'd want ?  If I were in your shoes I'd print out, say, the request source name at the very least.  Assuming it crashes on the next line.

Reply:
that's a great point. will add